### PR TITLE
[DebugInfo][InstrRef][MIR][GlobalIsel][MachineLICM] Use std::move to avoid copying

### PR DESCRIPTION
**Pull request by abhishek-kaushik22**

### CODE REVIEW

**Comment by abhishek-kaushik22** [comment]
@davemgreen @optimisan Can you please review?

**Comment by abhishek-kaushik22** [comment]
@kazutakahirata can you please review as well?

**Review by kazutakahirata — APPROVED** [review]
LGTM. Thank you for fixing these!

File: llvm/lib/CodeGen/MIRPrinter.cpp [hunk 1]
```diff
@@ -364,7 +364,7 @@ void MIRPrinter::convert(yaml::MachineFunction &YamlMF,
     for (const MCPhysReg *I = CalleeSavedRegs; *I; ++I) {
       yaml::FlowStringValue Reg;
       printRegMIR(*I, Reg, TRI);
-      CalleeSavedRegisters.push_back(Reg);
+      CalleeSavedRegisters.push_back(std::move(Reg));
     }
     YamlMF.CalleeSavedRegisters = CalleeSavedRegisters;
```
Comment (optimisan) [inline llvm/lib/CodeGen/MIRPrinter.cpp:369]: Can use `std::move` here as well right?

**Review by optimisan — APPROVED** [review]

**Comment by abhishek-kaushik22** [comment]
Can someone please merge?

**Pull request merged**